dspace-api/src/main/java/org/dspace/core/DBConnection.java

```diff
@@ -13,48 +13,121 @@
+     */
     public void closeDBConnection() throws SQLException;
 
-    public void commit() throws SQLException;
-
+    /**
+     * Close all sessions.  Release all associated resources (cache, DBMS
+     * connections, etc.)  To be used only when exiting the application.
+     */
     public void shutdown();
 
+    /**
+     * Some description of the DBMS used to persist entities.
+     * @return Brand, version, dialect, etc.  Implementation specific.
+     */
     public String getType();
 
+    /**
+     * The JDBC DataSource used by this session.  Think carefully before using.
+     * @return the source of DBMS connections.
+     */
     public DataSource getDataSource();
 
+    /**
+     * Identify certain characteristics of the DBMS being used to support persistence.
+     * @return a collection of DBMS and database information.
+     * @throws SQLException passed through.
+     */
     public DatabaseConfigVO getDatabaseConfig() throws SQLException;
-    
+
+    /**
+     * Configure the connection for special uses.
+     * @param batchOptimized if true, optimize for batch use.  Typically this
+     *  means suppressing automatic flushing of updates, thus requiring manual
+     *  flushing at appropriate points in the process.
+     * @param readOnlyOptimized if true, optimize for read-only use.  Typically
+     *  this suppresses all updating.
+     * @throws SQLException
+     */
     public void setConnectionMode(boolean batchOptimized, boolean readOnlyOptimized) throws SQLException;
 
+    /**
+     * Has this session been configured for large batches?  Typically this means
+     * that automatic flushing of updates to the database is suppressed, and
+     * thus one must take care to flush manually (or commit) at appropriate times.
+     * @return true if configured for batch.
+     */
     public boolean isOptimizedForBatchProcessing();
 
+    /**
+     * How many entities are cached in this session?
+     * @return number of cached entities.
+     * @throws SQLException passed through.
+     */
     public long getCacheSize() throws SQLException;
 
     /**
-     * Reload a DSpace object from the database. This will make sure the object is valid and stored in the cache.
+     * Reload a DSpace object from the database. This will make sure the object
+     * is valid and stored in the cache.  The returned object should be used
+     * henceforth instead of the passed object.
+     * @param <E> type of {@link entity}
      * @param entity The DSpace object to reload
+     * @return the reloaded entity.
+     * @throws java.sql.SQLException passed through.
      */
     public <E extends ReloadableEntity> E reloadEntity(E entity) throws SQLException;
 
     /**
-     * Remove a DSpace object from the cache when batch processing a large number of objects.
-     * @param entity The DSpace object to reload
+     * Remove a DSpace object from the session cache when batch processing a
+     * large number of objects.
```

Comment:
Minor note. Assuming `uncacheEntity` stays as-is, simply a tool to remove objects from session cache, we may want to add a warning here to note that "Objects removed from cache are not saved in any way. Therefore, if you have modified an object, you should be sure to commit() changes before calling this method"  (or something similar)